server/src/query/mod.rs

```diff
@@ -0,0 +1,545 @@
+                        )),
+                        Operator::Eq => Some((
+                            new_binary_expr(
+                                new_binary_expr(
+                                    logical_expr::col("start_time"),
+                                    Operator::LtEq,
+                                    *right.clone(),
+                                ),
+                                Operator::And,
+                                new_binary_expr(
+                                    logical_expr::col("end_time"),
+                                    Operator::GtEq,
+                                    *right.clone(),
+                                ),
+                            ),
+                            new_binary_expr(logical_expr::col("timestamp"), *op, *right.clone()),
+                        )),
+                        _ => None,
+                    }
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        }
+        _other => None,
+    }
+}
+
+/// Create a [`Expr::BinaryExpr`].
+fn new_binary_expr(left: Expr, op: Operator, right: Expr) -> Expr {
+    Expr::BinaryExpr(BinaryExpr {
+        left: Box::new(left),
+        op,
+        right: Box::new(right),
+    })
+}
+
+#[async_trait]
+impl TableProvider for ModelTable {
+    /// Return `self` as [`Any`] so it can be downcast.
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    /// Return the schema of the model table registered with Apache Arrow DataFusion.
+    fn schema(&self) -> SchemaRef {
+        self.model_table_metadata.schema.clone()
+    }
+
+    /// Specify that model tables are base tables and not views or temporary.
+    fn table_type(&self) -> TableType {
+        TableType::Base
+    }
+
+    /// Specify that model tables performs inexact predicate push-down.
+    fn supports_filter_pushdown(&self, _filter: &Expr) -> Result<TableProviderFilterPushDown> {
+        Ok(TableProviderFilterPushDown::Inexact)
+    }
+
+    /// Create an [`ExecutionPlan`] that will scan the table. Returns a
+    /// [`DataFusionError::Plan`] if the necessary metadata cannot be retrieved
+    /// from the metadata database.
+    async fn scan(
+        &self,
+        ctx: &SessionState,
+        projection: Option<&Vec<usize>>,
+        filters: &[Expr],
+        limit: Option<usize>,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        let table_name = &self.model_table_metadata.name.as_str();
+        let schema = &self.model_table_metadata.schema;
+
+        // Ensures a projection is present for looking up the columns to return.
+        let projection: Vec<usize> = if let Some(projection) = projection {
+            projection.to_vec()
+        } else {
+            (0..schema.fields().len()).collect()
+        };
+
+        // So SortedJoinExec simply needs to append arrays to a vector, the order of the field and
```

Comment:
Should it be `Since` instead of `So`?

Reply:
Fixed and changed `SortedJoinExec` to `SortedJoinStream` as it is the streams that perform the described actions.